sonar/modules/documents/api.py

```diff
@@ -181,19 +181,25 @@ def sync_files(self, file, deleted=False):
         :param file: File object.
         :param deleted: Wether the given file has been deleted or not.
         """
-        # For documents, a thumbnail and a fulltext file is generated.
+        # Synchronise files between bucket and record.
+        self.files.flush()
+
+        # If file is not deleted, a thumbnail and a fulltext file is generated.
         if not deleted:
             self.create_fulltext_file(self.files[file.key])
             self.create_thumbnail(self.files[file.key])
```

Comment:
Why not check if the file is already generated?

Reply:
This method is called when a file is uploaded or deleted, so I think it's necessary to create the thumbnail for the new file.